zebedee-cms/src/test/java/com/github/onsdigital/zebedee/api/CollectionTest.java

```diff
@@ -246,4 +269,197 @@ public void testPost_permissionDenied_shouldThrowException() throws Exception {
+                .thenReturn(session);
+
+        when(mockRequest.getPathInfo())
+                .thenReturn("collections/" + COLLECTION_ID);
+
+        when(collections.getCollection(COLLECTION_ID))
+                .thenReturn(collection);
+
+        when(collection.getId())
+                .thenReturn(COLLECTION_ID);
+
+        when(usersService.getUserByEmail(TEST_EMAIL))
+                .thenReturn(null);
+
+        NotFoundException ex = assertThrows(NotFoundException.class,
+                () -> endpoint.deleteCollection(mockRequest, mockResponse));
+
+        assertThat(ex.getMessage(), equalTo(format("requested user was not found: {0}", TEST_EMAIL)));
+        verify(sessions, times(1)).get(mockRequest);
+        verify(collections, times(1)).getCollection(COLLECTION_ID);
+        verify(collections, times(1)).delete(collection, session);
+        verify(usersService, times(1)).getUserByEmail(TEST_EMAIL);
+    }
+
+    @Test
+    public void testDelete_keyringRemoveError_shouldThrowEx() throws Exception {
+        when(sessions.get(mockRequest))
+                .thenReturn(session);
+
+        when(mockRequest.getPathInfo())
+                .thenReturn("collections/" + COLLECTION_ID);
+
+        when(collections.getCollection(COLLECTION_ID))
+                .thenReturn(collection);
+
+        when(collection.getId())
+                .thenReturn(COLLECTION_ID);
+
+        when(usersService.getUserByEmail(TEST_EMAIL))
+                .thenReturn(user);
+
+        doThrow(KeyringException.class)
+                .when(keyring)
+                .remove(user, collection);
+
+        InternalServerError ex = assertThrows(InternalServerError.class,
+                () -> endpoint.deleteCollection(mockRequest, mockResponse));
+
+        assertThat(ex.getMessage(), equalTo(format("error attempting to remove collection key from keyring: {0}", COLLECTION_ID)));
+        verify(sessions, times(1)).get(mockRequest);
+        verify(collections, times(1)).getCollection(COLLECTION_ID);
+        verify(collections, times(1)).delete(collection, session);
+        verify(usersService, times(1)).getUserByEmail(TEST_EMAIL);
+        verify(keyring, times(1)).remove(user, collection);
+    }
+
+    @Test
+    public void testDelete_success_shouldDeleteCollection() throws Exception {
+        when(sessions.get(mockRequest))
+                .thenReturn(session);
+
+        when(mockRequest.getPathInfo())
+                .thenReturn("collections/1234");
+
+        when(collections.getCollection("1234"))
+                .thenReturn(collection);
+
+        when(usersService.getUserByEmail(TEST_EMAIL))
+                .thenReturn(user);
+
+        boolean result = endpoint.deleteCollection(mockRequest, mockResponse);
+
+        assertTrue(result);
+        verify(collections, times(1)).delete(collection, session);
+        verify(scheduleCanceller, times(1)).cancel(collection);
+        verify(keyring, times(1)).remove(user, collection);
+    }
```

Comment:
Do we need to add test for schedule canceller too?

Reply:
The actual cancel functionality is done in Root and is existing functionality. I've added an interface abstraction around it to make it easier to test this class. I've taken the decision to not test that code as its beyond the scope of the change I am making here.